src/telemetry/activator.ts

```diff
@@ -19,41 +19,64 @@
     const reporter = ballerinaExtInstance.telemetryReporter;
     const langClient = <ExtendedLangClient>ballerinaExtInstance.langClient;
 
     // Start listening telemtry events from language server
-    ballerinaExtInstance.onReady().then(() => {
-        langClient.onNotification('telemetry/event', (event: LSErrorTelemetryEvent | LSTelemetryEvent) => {
-            if ('errorMessage' in event) {
-                const errorEvent: LSErrorTelemetryEvent = event;
-                const props: { [key: string]: string; } = getTelemetryProperties(ballerinaExtInstance, event.component, event.message)
-                props["ballerina.langserver.error.stacktrace"] = errorEvent.errorStackTrace;
-                props["ballerina.langserver.error.message"] = errorEvent.errorMessage;
-                reporter.sendTelemetryEvent(TM_ERROR_LANG_SERVER, props);
-            } else {
-                reporter.sendTelemetryEvent(TM_EVENT_LANG_SERVER, getTelemetryProperties(ballerinaExtInstance, event.component, event.message));
-            }
-        });
-    })
+    ballerinaExtInstance.onReady()
+        .then(() => {
+            langClient.onNotification('telemetry/event', (event: LSTelemetryEvent) => {
+                let props: { [key: string]: string; };
+                switch (event.type) {
+                    case TM_EVENT_TYPE_ERROR:
+                        const errorEvent: LSErrorTelemetryEvent = <LSErrorTelemetryEvent>event;
+                        props = getTelemetryProperties(ballerinaExtInstance, event.component, errorEvent.message);
+                        props["ballerina.langserver.error.description"] = errorEvent.message;
+                        props["ballerina.langserver.error.stacktrace"] = errorEvent.errorStackTrace;
+                        props["ballerina.langserver.error.message"] = errorEvent.errorMessage;
+                        props["ballerina.langserver.event"] = JSON.stringify(event);
+                        reporter.sendTelemetryEvent(TM_ERROR_LANG_SERVER, props);
+                        break;
+                    case TM_EVENT_TYPE_FEATURE_USAGE:
+                        const usageEvent: LSFeatureUsageTelemetryEvent = <LSFeatureUsageTelemetryEvent>event;
+                        props = getTelemetryProperties(ballerinaExtInstance, event.component, usageEvent.featureName);
+                        props["ballerina.langserver.feature.name"] = usageEvent.featureName;
+                        props["ballerina.langserver.feature.class"] = usageEvent.featureClass;
+                        props["ballerina.langserver.feature.message"] = usageEvent.featureMessage;
+                        props["ballerina.langserver.event"] = JSON.stringify(event);
+                        reporter.sendTelemetryEvent(TM_FEATURE_USAGE_LANG_SERVER, props);
+                        break;
+                    default:
+                        props = getTelemetryProperties(ballerinaExtInstance, event.component, event.type);
```

Comment:
What type of telemetry events falls under default? The initial plan was to send a predefined set of events from LS to the plugin.

Reply:
As of now, there's no event type which falls into default category. But, when a new version of the plugin (after merging this) works with an older version of the LS (SLA4 or earlier), the `event.type` property will be `undefined` causing it to fall into `default` case. That's why I kept it here. 

We can actually ignore that and remove the `default` case as well. WDYT?

Reply:
It's okay to keep if it won't result in errors for non-predefined cases (Cannot afford to have any errors while sending telemetry). +1 to remove.

Reply:
Thanks! Removed the `default` case as discussed offline.